docs/sso.md

```diff
@@ -0,0 +1,111 @@
+# Single Sign-On
+
+The consumerfinance.gov codebase support single sign-on (SSO) with [Open ID Connect (OIDC) providers](https://openid.net/). We use Mozilla's [mozilla-django-oidc](https://mozilla-django-oidc.readthedocs.io/en/stable/) library for OIDC support.
+
+## Local implementation
+
+Our implementation of OIDC SSO uses a custom subclass of mozilla-django-oidc's `OIDCAuthenticationBackend` that supports additional OIDC claims processing. This enables us to map OIDC roles to Django groups and/or superuser permissioning, and lets us synchronize a user's name to the profile information in the OIDC provider.
+
+This claims processing is implemented via a Django setting that maps an [ODIC claim](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims) to a Python callback that will process it. The callable makes what changes it needs to and return `True` if the user has been modified and should be saved. Our `OIDC_CLAIMS_PROCESSORS` setting looks like this:
+
+```python
+OIDC_CLAIMS_PROCESSORS = {
+    "roles": "login.auth.process_roles_claim",
+    "given_name": "login.auth.process_given_name_claim",
+    "family_name": "login.auth.process_family_name_claim",
+}
+```
+
+Here, `family_name` is processed with `process_family_name_claim`, which is implemented as:
+
+```python
+def process_family_name_claim(user, family_name):
+    if user.last_name != family_name:
+        user.last_name = family_name
+        return True
+```
+
+For user roles, we have additional settings. We distinguish two user roles: one for admins, which maps to Django's `is_superuser`, and one for regular users, who are allowed to log into the Wagtail admin. For each there is a setting for the OIDC role:
+
+- `OIDC_OP_ADMIN_ROLE`
+- `OIDC_OP_USER_ROLE`
+
+And there is a setting for the name of the default group that all `OIDC_OP_USER_ROLE` users should belong to:
+
+- `SSO_DEFAULT_GROUP`
+
+If this group does not exist, it will be created. If it does include the `wagtailadmin.access_admin`, that permission will be added.
```

Comment:
```suggestion
If this group does not exist, it will be created. If the group already exists but does not include the `wagtailadmin.access_admin` permission, that permission will be added.
```